great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py

```diff
@@ -118,6 +118,42 @@ def get_attributed_metrics_by_domain(
         }
         return metrics_attributed_values_by_domain
 
+    def filter_expectation_configurations_by_column_names(
+        self,
+        include_column_names: Optional[List[str]] = None,
+        exclude_column_names: Optional[List[str]] = None,
+    ) -> None:
+        """Modify the expectation configuration attribute of the result object based on input column lists (performed in-place).
+
+        Args:
+            include_column_names: Explicitly specified desired columns (if None, it is computed based on active Batch).
+            exclude_column_names: If provided, these columns are pre-filtered and excluded from consideration.
+
+        Raises:
+            ValueError if both include_column_names and exclude_column_names have been provided.
+        """
+        if include_column_names is not None and exclude_column_names is not None:
+            raise ValueError(
+                "You may either use `include_column_names` or `exclude_column_names` (but not both)."
+            )
+
+        if self.expectation_configurations is None:
+            return
+
+        filtered_expectation_configurations: List[ExpectationConfiguration] = []
+
+        for expectation_configuration in self.expectation_configurations:
```

Comment:
@cdkini We also need to apply filtering to the `metrics_by_domain` property (whereby the `domain_type` field of the `Domain` objects, used as keys, may have the `column` field defined).  Some thinking may need to be applied to two-column and multi-column domains (and the same for the `ExpectationConfiguration` filtering, already in place.  Thank you!